src/backend/catalog/pg_depend.c

```diff
@@ -91,6 +94,13 @@ recordMultipleDependencies(const ObjectAddress *depender,
 		 */
 		if (!isObjectPinned(referenced, dependDesc))
 		{
+			/*
+			 * Lock the referenced object to protect it from dropping by
+			 * another transaction.
+			 */
+			if (behavior == DEPENDENCY_NORMAL || behavior == DEPENDENCY_AUTO)
```

Comment:
Do you have test for DEPENDENCY_AUTO?

Is it expected?
```sql
-- current session:
gpadmin=# create sequence test_22_seq;
CREATE SEQUENCE
gpadmin=# begin;
BEGIN
gpadmin=# create table test_22_table(id int default nextval('test_22_seq'));
CREATE TABLE

-- other session: 
drop sequence test_22_seq; 

-- current session:
gpadmin=# commit;
COMMIT
gpadmin=# insert into test_22_table default values;
ERROR:  relation not found (OID 19102)  (entry db 172.18.0.2:6000 pid=54292)
DETAIL:  This can be validly caused by a concurrent delete operation on this object.
gpadmin=# select * from test_22_table;
 id 
----
(0 rows)
```